tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -767,115 +767,161 @@ def test_constrained_sample_independent_zero_startup() -> None:
     sampler.sample_independent(study, trial, "param-a", dist)
 
 
-@pytest.mark.parametrize("direction", ["minimize", "maximize"])
-@pytest.mark.parametrize("constraints_enabled", [False, True])
-def test_get_observation_pairs(direction: str, constraints_enabled: bool) -> None:
-    def objective(trial: Trial) -> float:
+def get_observation_pairs_objective(trial: Trial) -> float:
```

Comment:
How about moving this function inside `test_get_obdervation_pairs`? And, I'm not sure, but we should define other function for `test_get_observation_pairs_constrained` to set the constraint values.